flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/PartitionPathUtils.java

```diff
@@ -88,7 +88,9 @@ public static String generatePartitionPath(LinkedHashMap<String, String> partiti
 			suffixBuf.append(escapePathName(e.getValue()));
 			i++;
 		}
-		suffixBuf.append(Path.SEPARATOR);
+		if (partitionSpec.size() > 0) {
```

Comment:
we can move the judgement to the begin to make the code path short

Reply:
Good catch.
```
if (partitionSpec.isEmpty()) {
  return "";
}
```